### PR TITLE
Todo update

### PR DESCRIPTION
**Pull request by aprzn123**
I've made what is at least a draft of the revamp of todo functionality I've been working on. Todo items can now be given tags and filtered by tag; I've also updated the progress value to be a percentage, and allowed the user to filter out completed tasks.

Based on issue #758

### CODE REVIEW

File: jarviscli/plugins/reminder.py [hunk 1]
```diff
@@ -380,13 +560,87 @@ class Todo_Progress(TodoBase):
...
+class Todo_Tag(TodoBase):
+    """
+    Add a tag to an item which lets you filter later.
+    """
+    def __call__(self, jarvis, s):
+        entry = self.select_one_remind(jarvis)
+        tags = TagBase()
+        selected_tag = tags.select_one_tag(jarvis)
+        try:
+            entry["tags"].append(selected_tag["id"])
```
Comment (pnhofmann) [inline jarviscli/plugins/reminder.py:582]: Will fail if no tag exists
```
Some error occurred, please open an issue on github!
Here is error:

Traceback (most recent call last):
  File "jarviscli/CmdInterpreter.py", line 287, in try_do
    do(self, s)
  File "jarviscli/plugin.py", line 216, in run
    self.get_plugins(command).run(jarvis, new_s)
  File "jarviscli/plugin.py", line 208, in run
    self._backend[0](jarvis.get_api(), s)
  File "/home/phil/git/Jarvis/jarviscli/plugins/reminder.py", line 582, in __call__
    entry["tags"].append(selected_tag["id"])
TypeError: 'NoneType' object is not subscriptable
```

File: jarviscli/plugins/reminder.py [hunk 2]
```diff
@@ -23,11 +23,124 @@
...
+            return
+
+        # open selection menu
+        ask_str = []
+        for entry in data:
+            ask_str.append(self.format(jarvis, entry))
+
+        title = 'Please choose tag to remove (select with space, j and k to move up and down)'
+        selected = pick(ask_str, title, multi_select=True)
+        print(selected)
```
Comment (pnhofmann) [inline jarviscli/plugins/reminder.py:131]: Why this print?

File: jarviscli/plugins/reminder.py [hunk 3]
```diff
@@ -380,13 +560,87 @@ class Todo_Progress(TodoBase):
...
+        self.do_print(jarvis)
+
+@alias('tags create', 'tags add')
+@plugin('tags new')
+class Tags_New(TagBase):
+    """
+    Create a new tag.
+    """
+    def __call__(self, jarvis, s):
+        self.add_tag(jarvis, s)
```
Comment (pnhofmann) [inline jarviscli/plugins/reminder.py:624]: Maybe check s != '' - because empty string tags (created when just calling tags create) might be confusing ;).

**Review by pnhofmann — COMMENTED** [review]
```
flake8 --select E,W --max-line-length=140 --ignore E722,W503,W504,E128 jarviscli/plugins/reminder.py
jarviscli/plugins/reminder.py:28:72: W291 trailing whitespace
jarviscli/plugins/reminder.py:29:75: W291 trailing whitespace
jarviscli/plugins/reminder.py:33:1: E302 expected 2 blank lines, found 1
jarviscli/plugins/reminder.py:44:1: W293 blank line contains whitespace
jarviscli/plugins/reminder.py:47:1: W293 blank line contains whitespace
jarviscli/plugins/reminder.py:63:1: W293 blank line contains whitespace
jarviscli/plugins/reminder.py:85:1: W293 blank line contains whitespace
jarviscli/plugins/reminder.py:88:1: W293 blank line contains whitespace
jarviscli/plugins/reminder.py:93:1: W293 blank line contains whitespace
jarviscli/plugins/reminder.py:260:29: W291 trailing whitespace
jarviscli/plugins/reminder.py:274:96: W291 trailing whitespace
jarviscli/plugins/reminder.py:275:43: W291 trailing whitespace
jarviscli/plugins/reminder.py:300:1: W293 blank line contains whitespace
jarviscli/plugins/reminder.py:308:1: W293 blank line contains whitespace
jarviscli/plugins/reminder.py:310:1: W293 blank line contains whitespace
jarviscli/plugins/reminder.py:568:1: W293 blank line contains whitespace
jarviscli/plugins/reminder.py:617:1: E302 expected 2 blank lines, found 1
```

But looks good!

File: jarviscli/plugins/reminder.py (hunk 2 above)
Comment (aprzn123) [reply jarviscli/plugins/reminder.py:131]: forgot to delete some prints that were for testing purposes

**Comment by aprzn123** [comment]
I've fixed the things you pointed out, and hopefully didn't introduce any new bugs in the process.

**Comment by pnhofmann** [comment]
Thanks!

**Pull request merged**